core/src/main/scala/com/salesforce/op/filters/RawFeatureFilter.scala

```diff
@@ -371,6 +355,156 @@ object RawFeatureFilter {
     bins
   }
 
+  private def getAllSummaries(bins: Int, allFeatures: RDD[AllFeatures]): AllSummaries = {
+    def apply(sum: AllSummaries, feat: AllFeatures): AllSummaries = {
+      val (totalCount, responseSummaries, numericSummaries, textSummaries) = sum
+      val (responseFeatures, numericFeatures, textFeatures) = feat
+
+      def updateNumericSummaries(
+        summaries: Map[FeatureKey, HistogramSummary],
+        features: Map[FeatureKey, Seq[Double]]): Map[FeatureKey, HistogramSummary] =
+        summaries.keySet.union(features.keySet).map { key =>
+          val summary = summaries.get(key).getOrElse(new HistogramSummary(bins, bins * 10))
+          val points = features.get(key).getOrElse(Seq())
+
+          if (points.nonEmpty) key -> summary.update(points) else key -> summary
+        }.toMap
+
+      def updateTextSummaries(
```

Comment:
how would text bins formula be passed in to this?

Reply:
I think we should add textBinsFormula as input argument to both `getAllSummaries` and `updateTextSummaries`.